multiqc_cmgg/modules/msi_sensor_pro/msi_sensor_pro.py

```
@@ -107,6 +235,85 @@ def __init__(self):
+
+        return css_class, str(value)
+
+    def log_all_loci_stats(self, data_dicts_all, headers2):
+        try:
+            num_samples = len(data_dicts_all)
+            num_loci = len(headers2)
+            total_cells = sum(len(v) for v in data_dicts_all.values())
+            log.info(
+                f"Writing all-loci data: samples={num_samples}, loci={num_loci}, cells={total_cells}"
+            )
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)